rewrite-java/src/main/java/org/openrewrite/java/AddNullMethodArgument.java

```diff
@@ -0,0 +1,125 @@
+            example = "com.yourorg.A foo(int, int)")
+    String methodPattern;
+
+    /**
+     * A zero-based index that indicates which argument will be added as null to the method invocation.
+     */
+    @Option(displayName = "Argument index",
+            description = "A zero-based index that indicates which argument will be added as null to the method invocation.",
+            example = "0")
+    int argumentIndex;
+
+    @Override
+    public String getInstanceNameSuffix() {
+        return String.format("%d in methods `%s`", argumentIndex, methodPattern);
+    }
+
+    @Override
+    public String getDisplayName() {
+        return "Delete method argument";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Delete an argument from method invocations.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return Preconditions.check(new UsesMethod<>(methodPattern), new AddNullMethodArgumentVisitor(new MethodMatcher(methodPattern)));
+    }
+
+    private class AddNullMethodArgumentVisitor extends JavaIsoVisitor<ExecutionContext> {
+        private final MethodMatcher methodMatcher;
+
+        public AddNullMethodArgumentVisitor(MethodMatcher methodMatcher) {
+            this.methodMatcher = methodMatcher;
+        }
+
+        @Override
+        public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext ctx) {
+            J.MethodInvocation m = super.visitMethodInvocation(method, ctx);
+            return (J.MethodInvocation) visitMethodCall(m);
+        }
+
+        @Override
+        public J.NewClass visitNewClass(J.NewClass newClass, ExecutionContext ctx) {
+            J.NewClass n = super.visitNewClass(newClass, ctx);
+            return (J.NewClass) visitMethodCall(n);
+        }
+
+        private MethodCall visitMethodCall(MethodCall methodCall) {
+            MethodCall m = methodCall;
+            List<Expression> originalArgs = m.getArguments();
+            if (methodMatcher.matches(m) && (long) originalArgs.size() >= argumentIndex) {
+                List<Expression> args = new ArrayList<>(originalArgs);
+
+                if(args.size() == 1 && args.get(0) instanceof J.Empty) {
+                    args.remove(0);
+                }
+
+                args.add(argumentIndex, new J.Literal(randomId(), args.isEmpty() ? Space.EMPTY : Space.SINGLE_SPACE, Markers.EMPTY, "null", "null", null, JavaType.Primitive.Null));
+                m = m.withArguments(args);
+
+                JavaType.Method methodType = m.getMethodType();
+                if (methodType != null) {
+                    List<String> parameterNames = new ArrayList<>(methodType.getParameterNames());
+                    parameterNames.add(argumentIndex, "null");
+                    List<JavaType> parameterTypes = new ArrayList<>(methodType.getParameterTypes());
+                    parameterTypes.add(argumentIndex, JavaType.Primitive.Null);
```

Comment:
I'm not so sure about this parameter type `Null` here; Perhaps we need another argument where folks can pass in a FQN of the target type. That way any subsequent recipes in a larger migration will also match the method invocation going forward, especially if this is one partial step of a larger migration.

By comparison, now we end up with this after a recipe execution in `void addToMiddleArgument()`
![image](https://github.com/openrewrite/rewrite/assets/1027334/19ebce04-3b89-4de2-9d60-df9add68c317)

Reply:
ah, now this makes a lot more sense indeed. I did not fully understand what this part was doing to be honest 😅 
I think passing the type (maybe also param name?) is OK as you should always know the exact method you are migrating to. 
I don't think there is a way to determine/find this data if the method is not defined in the project is there?

Reply:
Not as easily reliably no, especially when we start stacking migration recipes where you start out without that new method available in an older version of a library.